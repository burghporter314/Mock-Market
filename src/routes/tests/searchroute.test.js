import React, { useState } from 'react';
import renderer from 'react-test-renderer';
import { useAuth0 } from '@auth0/auth0-react';
import SearchPage from '../Search';

jest.mock("@auth0/auth0-react");

beforeEach(() => {
    useAuth0.mockReturnValue({
        isAuthenticated: true,
        nickname: "testuser",
        logout: jest.fn(),
        loginWithRedirect: jest.fn()
    })
})

it('should match snapshot', () => {

    const component = renderer.create(
            <SearchPage/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})