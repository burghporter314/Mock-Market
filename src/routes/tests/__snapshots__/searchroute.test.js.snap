// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Array [
  <br />,
  <br />,
  <div
    className="container"
  >
    <h1 />
    <div
      className="mb-3 input-group"
    >
      <span
        className="input-group-text"
        id="basic-addon2"
      >
        Stock Ticker
      </span>
      <input
        aria-describedby="basic-addon2"
        aria-label="Enter Stock Ticker"
        className="form-control"
        id="inputTicker"
        placeholder="Enter Stock Ticker"
      />
      <button
        className="btn btn-outline-success"
        disabled={false}
        id="search"
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
    <br />
    <table
      className="table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            Stock Ticker
          </th>
          <th>
            Name
          </th>
          <th>
            Type
          </th>
          <th>
            Primary Exchange
          </th>
          <th>
            Locale
          </th>
          <th>
            Currency
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
  <div
    className="toast-container position-absolute bottom-0 start-50 translate-middle-x p-3"
  />,
]
`;
